### PR TITLE
Make bus coordinator information through json file configure.

### PR DESCRIPTION
**Pull request by BhagyashreeBhoyar**

### CODE REVIEW

File: src/components/SelectedStudentsActionWrapper.jsx [hunk 1]
```diff
@@ -30,8 +35,45 @@ class SelectedStudentsActionWrapper extends Component {
...
   printCards() {
     window.print();
+    this.setStateOfCoordinatorErrorPopup(false);
   }
 
+  /**
+   * renderIsBusCoordinatorsDataLoadedErrorPopup render bus coordinator error popup
+   * @return {ReactComponent}
+   */
+  renderIsBusCoordinatorsDataLoadedErrorPopup = () => {
```
Comment (pepcus-pratik) [inline src/components/SelectedStudentsActionWrapper.jsx]:
```suggestion
  renderCoordinatorUnavailableWarningPopup = () => {
```

File: src/components/SelectedStudentsActionWrapper.jsx [hunk 2]
```diff
@@ -30,8 +35,45 @@ class SelectedStudentsActionWrapper extends Component {
...
+    if (this.props.isBusCoordinatorsDataLoadedError && this.state.isBusCoordinatorsError) {
+      return (
+        <Popup>
+          <h5>Coordinator information was failed so, Do you want to proceed?</h5>
```
Comment (pepcus-pratik) [inline src/components/SelectedStudentsActionWrapper.jsx]:
```suggestion
          <h5>Coordinator information is not available and will not be printed on cards. Do you want to proceed ?</h5>
```

File: src/components/SelectedStudentsActionWrapper.jsx [hunk 3]
```diff
@@ -30,8 +35,45 @@ class SelectedStudentsActionWrapper extends Component {
...
+
+      );
+    }
+    return null;
+  };
+  /**
+   * setStateOfCoordinatorErrorPopup set the boolean value of isBusCoordinatorsError
+   * @param {Boolean}value
+   */
+  setStateOfCoordinatorErrorPopup = (value) => {
```
Comment (pepcus-pratik) [inline src/components/SelectedStudentsActionWrapper.jsx]:
```suggestion
  onClickPrintCancel = (value) => {
```

File: src/components/SelectedStudentsActionWrapper.jsx [hunk 4]
```diff
@@ -82,7 +124,14 @@ class SelectedStudentsActionWrapper extends Component {
               </CSVLink>
             </div>
             <div className="buttonContainer">
-              <button className={this.renderPrintNowClassName()} onClick={this.printCards}>
+              <button
+                className={this.renderPrintNowClassName()}
```
Comment (pepcus-pratik) [inline src/components/SelectedStudentsActionWrapper.jsx]:
```suggestion
                className={this.getPrintNowClassName()}
```

File: src/components/SelectedStudentsActionWrapper.jsx [hunk 5]
```diff
@@ -82,7 +124,14 @@ class SelectedStudentsActionWrapper extends Component {
...
+                onClick={
+                  () => {
+                    this.props.isBusCoordinatorsDataLoadedError
```
Comment (pepcus-pratik) [inline src/components/SelectedStudentsActionWrapper.jsx]:
```suggestion
                    this.props.isCoordinatorsDataFailed
```

File: src/dataGrid.css [hunk 6]
```diff
@@ -419,6 +419,23 @@ html, body {
     border-radius: 4px !important;
 }
 
+.bus-coordinatorsErrorPopupButtonContainer button {
```
Comment (pepcus-pratik) [inline src/dataGrid.css]: incorrect class name convention

File: src/dataGrid.css [hunk 7]
```diff
@@ -419,6 +419,23 @@ html, body {
...
+    background-color: var(--enable-link-color) !important;
+    color: #000;
+    border: 2px solid var(--enable-link-color) !important;
+    width: 100px;
+    padding: 6px 0;
+}
+.bus-coordinatorsErrorPopupButtonContainer button:hover {
```
Comment (pepcus-pratik) [inline src/dataGrid.css]: incorrect class name convention

File: src/dataGrid.css [hunk 8]
```diff
@@ -419,6 +419,23 @@ html, body {
...
+    background: var(--enable-link-hover-color) !important;
+    border: 2px solid var(--enable-link-hover-color) !important;
+}
+.bus-coordinatorsErrorPopupButtonContainer {
```
Comment (pepcus-pratik) [inline src/dataGrid.css]: incorrect class name convention

File: src/reducers/assetFilesReducer.js [hunk 9]
```diff
@@ -52,6 +52,26 @@ export const assetFilesReducer = (state = initialState, action) => {
         isAppLoaded: false,
         isAppLoadedError: true,
       };
+    case 'LOAD_BUS_COORDINATORS_DATA_ACTION':
+      return {
+        ...state,
+        isBusCoordinatorsDataLoadedError: false,
+      };
+    case 'LOADED_BUS_COORDINATORS_DATA_SUCCESS_ACTION':
```
Comment (pepcus-pratik) [inline src/reducers/assetFilesReducer.js]:
```suggestion
    case 'LOAD_BUS_COORDINATORS_DATA_SUCCESS_ACTION':
```

File: src/reducers/assetFilesReducer.js [hunk 10]
```diff
@@ -70,4 +90,8 @@ export const getApplicationMode = state => state.assetFilesReducer.modeVariable;
 
 export const isAppLoaded = state => state.assetFilesReducer.isAppLoaded;
 
-export const getIsAppLoadedError = state => state.assetFilesReducer.isAppLoadedError;
\ No newline at end of file
+export const getIsAppLoadedError = state => state.assetFilesReducer.isAppLoadedError;
+
+export const getBusCoordinators = state => state.assetFilesReducer.busCoordinators;
+
+export const isBusCoordinatorsDataLoadedError = state => state.assetFilesReducer.isBusCoordinatorsDataLoadedError;
```
Comment (pepcus-pratik) [inline src/reducers/assetFilesReducer.js]:
```suggestion
export const isBusCoordinatorsDataFailed = state => state.assetFilesReducer.isBusCoordinatorsDataFailed;
```

File: src/sagas/assetFilesSaga.js [hunk 11]
```diff
@@ -37,3 +39,18 @@ export function* getAppConfigSaga() {
     yield put(loadAppDataFailedAction(errorMessage));
   }
 }
+
+export function* getBusCoordinatorsConfigSaga() {
+  const errorMessage = 'Unable to fetch  bus coordinators config.';
+  try {
+    const response = yield getBusCoordinatorsConfig();
+    if (response) {
+      yield put(loadedBusCoordinatorsDataSuccessAction(response));
```
Comment (pepcus-pratik) [inline src/sagas/assetFilesSaga.js]:
```suggestion
      yield put(loadBusCoordinatorsDataSuccessAction(response));
```

File: src/components/SelectedStudentsActionWrapper.jsx (hunk 1 above)
Comment (BhagyashreeBhoyar) [reply src/components/SelectedStudentsActionWrapper.jsx]: Done.

File: src/components/SelectedStudentsActionWrapper.jsx (hunk 2 above)
Comment (BhagyashreeBhoyar) [reply src/components/SelectedStudentsActionWrapper.jsx]: That message is converted into Hindi text.

File: src/components/SelectedStudentsActionWrapper.jsx (hunk 3 above)
Comment (BhagyashreeBhoyar) [reply src/components/SelectedStudentsActionWrapper.jsx]: Done.

File: src/components/SelectedStudentsActionWrapper.jsx (hunk 4 above)
Comment (BhagyashreeBhoyar) [reply src/components/SelectedStudentsActionWrapper.jsx]: Done.

File: src/components/SelectedStudentsActionWrapper.jsx (hunk 5 above)
Comment (BhagyashreeBhoyar) [reply src/components/SelectedStudentsActionWrapper.jsx]: Done.

File: src/dataGrid.css (hunk 7 above)
Comment (BhagyashreeBhoyar) [reply src/dataGrid.css]: Done.

File: src/dataGrid.css (hunk 8 above)
Comment (BhagyashreeBhoyar) [reply src/dataGrid.css]: Done.

File: src/reducers/assetFilesReducer.js (hunk 9 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/assetFilesReducer.js]: Done.

File: src/reducers/assetFilesReducer.js (hunk 10 above)
Comment (BhagyashreeBhoyar) [reply src/reducers/assetFilesReducer.js]: Done.

File: src/sagas/assetFilesSaga.js (hunk 11 above)
Comment (BhagyashreeBhoyar) [reply src/sagas/assetFilesSaga.js]: Done.

**Pull request merged**